opentelemetry-sdk/tests/trace/test_trace.py

```diff
@@ -641,16 +641,26 @@ def test_ended_span(self):
         )
 
     def test_error_status(self):
-        try:
-            with trace.TracerSource().get_tracer(__name__).start_span(
-                "root"
-            ) as root:
-                raise Exception("unknown")
-        except Exception:  # pylint: disable=broad-except
-            pass
-
-        self.assertIs(root.status.canonical_code, StatusCanonicalCode.UNKNOWN)
-        self.assertEqual(root.status.description, "Exception: unknown")
+        def error_status_test(context):
+            try:
+                with context as root:
+                    raise Exception("unknown")
+            except Exception:  # pylint: disable=broad-except
+                pass
+
+            self.assertIs(
+                root.status.canonical_code, StatusCanonicalCode.UNKNOWN
+            )
+            self.assertEqual(root.status.description, "Exception: unknown")
+
+        error_status_test(
```

Comment:
This should check that the exception escapes with `assertRaises`.

Also FWIW I think this would be clearer without `error_status_test`. In general I think it's more important for tests to be readable than DRY.

E.g.

```python
    def test_error_status(self):
        tracer = trace.TracerSource().get_tracer(__name__)

        ex = Exception("unknown")
        with self.assertRaises(Exception) as ec:
            with tracer.start_span("root") as span:
                raise ex

        self.assertIs(ex, ec.exception)
        self.assertIs(
            span.status.canonical_code, StatusCanonicalCode.UNKNOWN
        )
        self.assertEqual(span.status.description, "Exception: unknown")
        self.assertFalse(span.status.is_ok)

        with self.assertRaises(Exception) as ec:
            with tracer.start_as_current_span("root") as span:
                raise ex

        self.assertIs(ex, ec.exception)
        self.assertIs(
            span.status.canonical_code, StatusCanonicalCode.UNKNOWN
        )
        self.assertEqual(span.status.description, "Exception: unknown")
        self.assertFalse(span.status.is_ok)
```

Reply:
This is not only a matter of being DRY, there is something much more important. What the test does as it is now is that it quickly tells the reader another purpose of this test: _we are testing that `start_span` and `start_as_current_span` behave in the **same** way_. If we repeat our code we force our reader to carefully look into any possible differences between the two tests to make sure that these two methods are actually behaving in the same way.